Various configuration files.

Most notibly, the cached results of the xs-reconfigure.sh script.

diff --git a/.config b/.config
new file mode 100644
index 0000000000..6e67d2157c
--- /dev/null
+++ b/.config
@@ -0,0 +1,14 @@
+# -*- mode: Makefile; -*-
+
+# Always produce debug symbols for debuginfo RPMs
+debug_symbols := y
+
+# Enable all config options
+XEN_CONFIG_EXPERT := y
+export XEN_CONFIG_EXPERT
+
+# No tests
+CONFIG_TESTS := n
+
+# No remus netbuffering
+override CONFIG_REMUS_NETBUF := n
diff --git a/buildconfigs/config-debug b/buildconfigs/config-debug
new file mode 100644
index 0000000000..83da4c5f8c
--- /dev/null
+++ b/buildconfigs/config-debug
@@ -0,0 +1,119 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Xen/x86 4.13.2 Configuration
+#
+CONFIG_X86_64=y
+CONFIG_X86=y
+CONFIG_ARCH_DEFCONFIG="arch/x86/configs/x86_64_defconfig"
+
+#
+# Architecture Features
+#
+CONFIG_NR_CPUS=512
+CONFIG_PV=y
+CONFIG_PV_LINEAR_PT=y
+CONFIG_HVM=y
+CONFIG_SHADOW_PAGING=y
+# CONFIG_BIGMEM is not set
+CONFIG_HVM_FEP=y
+CONFIG_TBOOT=y
+CONFIG_XEN_ALIGN_DEFAULT=y
+# CONFIG_XEN_ALIGN_2M is not set
+# CONFIG_XEN_GUEST is not set
+# CONFIG_MEM_SHARING is not set
+
+#
+# Common Features
+#
+CONFIG_COMPAT=y
+CONFIG_CORE_PARKING=y
+CONFIG_GRANT_TABLE=y
+CONFIG_HAS_ALTERNATIVE=y
+CONFIG_HAS_EX_TABLE=y
+CONFIG_HAS_FAST_MULTIPLY=y
+CONFIG_MEM_ACCESS_ALWAYS_ON=y
+CONFIG_MEM_ACCESS=y
+CONFIG_HAS_MEM_PAGING=y
+CONFIG_HAS_PDX=y
+CONFIG_HAS_UBSAN=y
+CONFIG_HAS_KEXEC=y
+CONFIG_HAS_GDBSX=y
+CONFIG_HAS_IOPORTS=y
+CONFIG_HAS_SCHED_GRANULARITY=y
+CONFIG_NEEDS_LIBELF=y
+
+#
+# Speculative hardening
+#
+CONFIG_SPECULATIVE_HARDEN_ARRAY=y
+CONFIG_SPECULATIVE_HARDEN_BRANCH=y
+CONFIG_KEXEC=y
+# CONFIG_EFI_SET_VIRTUAL_ADDRESS_MAP is not set
+CONFIG_XENOPROF=y
+CONFIG_XSM=y
+# CONFIG_XSM_FLASK is not set
+CONFIG_XSM_SILO=y
+# CONFIG_XSM_DUMMY_DEFAULT is not set
+CONFIG_XSM_SILO_DEFAULT=y
+# CONFIG_LATE_HWDOM is not set
+# CONFIG_ARGO is not set
+
+#
+# Schedulers
+#
+CONFIG_SCHED_CREDIT=y
+CONFIG_SCHED_CREDIT2=y
+CONFIG_SCHED_RTDS=y
+CONFIG_SCHED_ARINC653=y
+CONFIG_SCHED_NULL=y
+CONFIG_SCHED_CREDIT_DEFAULT=y
+# CONFIG_SCHED_CREDIT2_DEFAULT is not set
+# CONFIG_SCHED_RTDS_DEFAULT is not set
+# CONFIG_SCHED_ARINC653_DEFAULT is not set
+# CONFIG_SCHED_NULL_DEFAULT is not set
+CONFIG_SCHED_DEFAULT="credit"
+CONFIG_CRYPTO=y
+CONFIG_LIVEPATCH=y
+CONFIG_FAST_SYMBOL_LOOKUP=y
+CONFIG_ENFORCE_UNIQUE_SYMBOLS=y
+CONFIG_CMDLINE=""
+CONFIG_DOM0_MEM=""
+CONFIG_TRACEBUFFER=y
+
+#
+# Device Drivers
+#
+CONFIG_ACPI=y
+CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
+CONFIG_NUMA=y
+CONFIG_HAS_NS16550=y
+CONFIG_HAS_EHCI=y
+CONFIG_HAS_CPUFREQ=y
+CONFIG_HAS_PASSTHROUGH=y
+CONFIG_HAS_PCI=y
+CONFIG_VIDEO=y
+CONFIG_VGA=y
+CONFIG_HAS_VPCI=y
+
+#
+# Deprecated Functionality
+#
+# CONFIG_PV_LDT_PAGING is not set
+CONFIG_DEFCONFIG_LIST="arch/x86/configs/x86_64_defconfig"
+CONFIG_ARCH_SUPPORTS_INT128=y
+
+#
+# Debugging Options
+#
+CONFIG_DEBUG=y
+# CONFIG_CRASH_DEBUG is not set
+CONFIG_DEBUG_INFO=y
+CONFIG_FRAME_POINTER=y
+# CONFIG_DEBUG_LOCK_PROFILE is not set
+CONFIG_DEBUG_LOCKS=y
+# CONFIG_PERF_COUNTERS is not set
+CONFIG_VERBOSE_DEBUG=y
+CONFIG_SCRUB_DEBUG=y
+# CONFIG_UBSAN is not set
+# CONFIG_DEBUG_TRACE is not set
+CONFIG_XMEM_POOL_POISON=y
diff --git a/buildconfigs/config-pvshim-debug b/buildconfigs/config-pvshim-debug
new file mode 100644
index 0000000000..7d6ba4a9fa
--- /dev/null
+++ b/buildconfigs/config-pvshim-debug
@@ -0,0 +1,109 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Xen/x86 4.13.2 Configuration
+#
+CONFIG_X86_64=y
+CONFIG_X86=y
+CONFIG_ARCH_DEFCONFIG="arch/x86/configs/x86_64_defconfig"
+
+#
+# Architecture Features
+#
+CONFIG_NR_CPUS=32
+CONFIG_PV=y
+CONFIG_PV_LINEAR_PT=y
+# CONFIG_HVM is not set
+# CONFIG_SHADOW_PAGING is not set
+# CONFIG_BIGMEM is not set
+# CONFIG_TBOOT is not set
+# CONFIG_XEN_ALIGN_DEFAULT is not set
+CONFIG_XEN_ALIGN_2M=y
+CONFIG_XEN_GUEST=y
+CONFIG_PVH_GUEST=y
+CONFIG_PV_SHIM=y
+CONFIG_PV_SHIM_EXCLUSIVE=y
+
+#
+# Common Features
+#
+CONFIG_COMPAT=y
+CONFIG_CORE_PARKING=y
+CONFIG_GRANT_TABLE=y
+CONFIG_HAS_ALTERNATIVE=y
+CONFIG_HAS_EX_TABLE=y
+CONFIG_HAS_FAST_MULTIPLY=y
+CONFIG_MEM_ACCESS_ALWAYS_ON=y
+CONFIG_MEM_ACCESS=y
+CONFIG_HAS_MEM_PAGING=y
+CONFIG_HAS_PDX=y
+CONFIG_HAS_UBSAN=y
+CONFIG_HAS_KEXEC=y
+CONFIG_HAS_GDBSX=y
+CONFIG_HAS_IOPORTS=y
+CONFIG_HAS_SCHED_GRANULARITY=y
+CONFIG_NEEDS_LIBELF=y
+
+#
+# Speculative hardening
+#
+CONFIG_SPECULATIVE_HARDEN_ARRAY=y
+CONFIG_SPECULATIVE_HARDEN_BRANCH=y
+# CONFIG_KEXEC is not set
+# CONFIG_EFI_SET_VIRTUAL_ADDRESS_MAP is not set
+# CONFIG_XENOPROF is not set
+# CONFIG_XSM is not set
+# CONFIG_ARGO is not set
+
+#
+# Schedulers
+#
+CONFIG_SCHED_CREDIT=y
+# CONFIG_SCHED_CREDIT2 is not set
+# CONFIG_SCHED_RTDS is not set
+# CONFIG_SCHED_ARINC653 is not set
+CONFIG_SCHED_NULL=y
+# CONFIG_SCHED_CREDIT_DEFAULT is not set
+CONFIG_SCHED_NULL_DEFAULT=y
+CONFIG_SCHED_DEFAULT="null"
+# CONFIG_LIVEPATCH is not set
+# CONFIG_ENFORCE_UNIQUE_SYMBOLS is not set
+# CONFIG_SUPPRESS_DUPLICATE_SYMBOL_WARNINGS is not set
+CONFIG_CMDLINE=""
+CONFIG_DOM0_MEM=""
+CONFIG_TRACEBUFFER=y
+
+#
+# Device Drivers
+#
+CONFIG_ACPI=y
+CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
+CONFIG_NUMA=y
+CONFIG_HAS_NS16550=y
+CONFIG_HAS_EHCI=y
+CONFIG_HAS_CPUFREQ=y
+CONFIG_HAS_PASSTHROUGH=y
+CONFIG_HAS_PCI=y
+
+#
+# Deprecated Functionality
+#
+# CONFIG_PV_LDT_PAGING is not set
+CONFIG_DEFCONFIG_LIST="arch/x86/configs/x86_64_defconfig"
+CONFIG_ARCH_SUPPORTS_INT128=y
+
+#
+# Debugging Options
+#
+CONFIG_DEBUG=y
+# CONFIG_CRASH_DEBUG is not set
+CONFIG_DEBUG_INFO=y
+CONFIG_FRAME_POINTER=y
+# CONFIG_COVERAGE is not set
+# CONFIG_DEBUG_LOCK_PROFILE is not set
+# CONFIG_DEBUG_LOCKS is not set
+# CONFIG_PERF_COUNTERS is not set
+# CONFIG_VERBOSE_DEBUG is not set
+# CONFIG_SCRUB_DEBUG is not set
+# CONFIG_UBSAN is not set
+# CONFIG_DEBUG_TRACE is not set
+# CONFIG_XMEM_POOL_POISON is not set
diff --git a/buildconfigs/config-pvshim-release b/buildconfigs/config-pvshim-release
new file mode 100644
index 0000000000..3464e1352c
--- /dev/null
+++ b/buildconfigs/config-pvshim-release
@@ -0,0 +1,109 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Xen/x86 4.13.2 Configuration
+#
+CONFIG_X86_64=y
+CONFIG_X86=y
+CONFIG_ARCH_DEFCONFIG="arch/x86/configs/x86_64_defconfig"
+
+#
+# Architecture Features
+#
+CONFIG_NR_CPUS=32
+CONFIG_PV=y
+CONFIG_PV_LINEAR_PT=y
+# CONFIG_HVM is not set
+# CONFIG_SHADOW_PAGING is not set
+# CONFIG_BIGMEM is not set
+# CONFIG_TBOOT is not set
+# CONFIG_XEN_ALIGN_DEFAULT is not set
+CONFIG_XEN_ALIGN_2M=y
+CONFIG_XEN_GUEST=y
+CONFIG_PVH_GUEST=y
+CONFIG_PV_SHIM=y
+CONFIG_PV_SHIM_EXCLUSIVE=y
+
+#
+# Common Features
+#
+CONFIG_COMPAT=y
+CONFIG_CORE_PARKING=y
+CONFIG_GRANT_TABLE=y
+CONFIG_HAS_ALTERNATIVE=y
+CONFIG_HAS_EX_TABLE=y
+CONFIG_HAS_FAST_MULTIPLY=y
+CONFIG_MEM_ACCESS_ALWAYS_ON=y
+CONFIG_MEM_ACCESS=y
+CONFIG_HAS_MEM_PAGING=y
+CONFIG_HAS_PDX=y
+CONFIG_HAS_UBSAN=y
+CONFIG_HAS_KEXEC=y
+CONFIG_HAS_GDBSX=y
+CONFIG_HAS_IOPORTS=y
+CONFIG_HAS_SCHED_GRANULARITY=y
+CONFIG_NEEDS_LIBELF=y
+
+#
+# Speculative hardening
+#
+CONFIG_SPECULATIVE_HARDEN_ARRAY=y
+CONFIG_SPECULATIVE_HARDEN_BRANCH=y
+# CONFIG_KEXEC is not set
+# CONFIG_EFI_SET_VIRTUAL_ADDRESS_MAP is not set
+# CONFIG_XENOPROF is not set
+# CONFIG_XSM is not set
+# CONFIG_ARGO is not set
+
+#
+# Schedulers
+#
+CONFIG_SCHED_CREDIT=y
+# CONFIG_SCHED_CREDIT2 is not set
+# CONFIG_SCHED_RTDS is not set
+# CONFIG_SCHED_ARINC653 is not set
+CONFIG_SCHED_NULL=y
+# CONFIG_SCHED_CREDIT_DEFAULT is not set
+CONFIG_SCHED_NULL_DEFAULT=y
+CONFIG_SCHED_DEFAULT="null"
+# CONFIG_LIVEPATCH is not set
+# CONFIG_ENFORCE_UNIQUE_SYMBOLS is not set
+# CONFIG_SUPPRESS_DUPLICATE_SYMBOL_WARNINGS is not set
+CONFIG_CMDLINE=""
+CONFIG_DOM0_MEM=""
+CONFIG_TRACEBUFFER=y
+
+#
+# Device Drivers
+#
+CONFIG_ACPI=y
+CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
+CONFIG_NUMA=y
+CONFIG_HAS_NS16550=y
+CONFIG_HAS_EHCI=y
+CONFIG_HAS_CPUFREQ=y
+CONFIG_HAS_PASSTHROUGH=y
+CONFIG_HAS_PCI=y
+
+#
+# Deprecated Functionality
+#
+# CONFIG_PV_LDT_PAGING is not set
+CONFIG_DEFCONFIG_LIST="arch/x86/configs/x86_64_defconfig"
+CONFIG_ARCH_SUPPORTS_INT128=y
+
+#
+# Debugging Options
+#
+# CONFIG_DEBUG is not set
+# CONFIG_CRASH_DEBUG is not set
+CONFIG_DEBUG_INFO=y
+# CONFIG_FRAME_POINTER is not set
+# CONFIG_COVERAGE is not set
+# CONFIG_DEBUG_LOCK_PROFILE is not set
+# CONFIG_DEBUG_LOCKS is not set
+# CONFIG_PERF_COUNTERS is not set
+# CONFIG_VERBOSE_DEBUG is not set
+# CONFIG_SCRUB_DEBUG is not set
+# CONFIG_UBSAN is not set
+# CONFIG_DEBUG_TRACE is not set
+# CONFIG_XMEM_POOL_POISON is not set
diff --git a/buildconfigs/config-release b/buildconfigs/config-release
new file mode 100644
index 0000000000..a6491fff5e
--- /dev/null
+++ b/buildconfigs/config-release
@@ -0,0 +1,119 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Xen/x86 4.13.2 Configuration
+#
+CONFIG_X86_64=y
+CONFIG_X86=y
+CONFIG_ARCH_DEFCONFIG="arch/x86/configs/x86_64_defconfig"
+
+#
+# Architecture Features
+#
+CONFIG_NR_CPUS=512
+CONFIG_PV=y
+CONFIG_PV_LINEAR_PT=y
+CONFIG_HVM=y
+CONFIG_SHADOW_PAGING=y
+# CONFIG_BIGMEM is not set
+CONFIG_HVM_FEP=y
+CONFIG_TBOOT=y
+CONFIG_XEN_ALIGN_DEFAULT=y
+# CONFIG_XEN_ALIGN_2M is not set
+# CONFIG_XEN_GUEST is not set
+# CONFIG_MEM_SHARING is not set
+
+#
+# Common Features
+#
+CONFIG_COMPAT=y
+CONFIG_CORE_PARKING=y
+CONFIG_GRANT_TABLE=y
+CONFIG_HAS_ALTERNATIVE=y
+CONFIG_HAS_EX_TABLE=y
+CONFIG_HAS_FAST_MULTIPLY=y
+CONFIG_MEM_ACCESS_ALWAYS_ON=y
+CONFIG_MEM_ACCESS=y
+CONFIG_HAS_MEM_PAGING=y
+CONFIG_HAS_PDX=y
+CONFIG_HAS_UBSAN=y
+CONFIG_HAS_KEXEC=y
+CONFIG_HAS_GDBSX=y
+CONFIG_HAS_IOPORTS=y
+CONFIG_HAS_SCHED_GRANULARITY=y
+CONFIG_NEEDS_LIBELF=y
+
+#
+# Speculative hardening
+#
+CONFIG_SPECULATIVE_HARDEN_ARRAY=y
+CONFIG_SPECULATIVE_HARDEN_BRANCH=y
+CONFIG_KEXEC=y
+# CONFIG_EFI_SET_VIRTUAL_ADDRESS_MAP is not set
+CONFIG_XENOPROF=y
+CONFIG_XSM=y
+# CONFIG_XSM_FLASK is not set
+CONFIG_XSM_SILO=y
+# CONFIG_XSM_DUMMY_DEFAULT is not set
+CONFIG_XSM_SILO_DEFAULT=y
+# CONFIG_LATE_HWDOM is not set
+# CONFIG_ARGO is not set
+
+#
+# Schedulers
+#
+CONFIG_SCHED_CREDIT=y
+CONFIG_SCHED_CREDIT2=y
+CONFIG_SCHED_RTDS=y
+CONFIG_SCHED_ARINC653=y
+CONFIG_SCHED_NULL=y
+CONFIG_SCHED_CREDIT_DEFAULT=y
+# CONFIG_SCHED_CREDIT2_DEFAULT is not set
+# CONFIG_SCHED_RTDS_DEFAULT is not set
+# CONFIG_SCHED_ARINC653_DEFAULT is not set
+# CONFIG_SCHED_NULL_DEFAULT is not set
+CONFIG_SCHED_DEFAULT="credit"
+CONFIG_CRYPTO=y
+CONFIG_LIVEPATCH=y
+CONFIG_FAST_SYMBOL_LOOKUP=y
+CONFIG_ENFORCE_UNIQUE_SYMBOLS=y
+CONFIG_CMDLINE=""
+CONFIG_DOM0_MEM=""
+CONFIG_TRACEBUFFER=y
+
+#
+# Device Drivers
+#
+CONFIG_ACPI=y
+CONFIG_ACPI_LEGACY_TABLES_LOOKUP=y
+CONFIG_NUMA=y
+CONFIG_HAS_NS16550=y
+CONFIG_HAS_EHCI=y
+CONFIG_HAS_CPUFREQ=y
+CONFIG_HAS_PASSTHROUGH=y
+CONFIG_HAS_PCI=y
+CONFIG_VIDEO=y
+CONFIG_VGA=y
+CONFIG_HAS_VPCI=y
+
+#
+# Deprecated Functionality
+#
+# CONFIG_PV_LDT_PAGING is not set
+CONFIG_DEFCONFIG_LIST="arch/x86/configs/x86_64_defconfig"
+CONFIG_ARCH_SUPPORTS_INT128=y
+
+#
+# Debugging Options
+#
+# CONFIG_DEBUG is not set
+# CONFIG_CRASH_DEBUG is not set
+CONFIG_DEBUG_INFO=y
+# CONFIG_FRAME_POINTER is not set
+# CONFIG_DEBUG_LOCK_PROFILE is not set
+# CONFIG_DEBUG_LOCKS is not set
+# CONFIG_PERF_COUNTERS is not set
+# CONFIG_VERBOSE_DEBUG is not set
+# CONFIG_SCRUB_DEBUG is not set
+# CONFIG_UBSAN is not set
+# CONFIG_DEBUG_TRACE is not set
+# CONFIG_XMEM_POOL_POISON is not set
diff --git a/config/Docs.mk b/config/Docs.mk
new file mode 100644
index 0000000000..6965e02475
--- /dev/null
+++ b/config/Docs.mk
@@ -0,0 +1,9 @@
+-include $(XEN_ROOT)/config/Paths.mk
+
+# Tools
+FIG2DEV             := 
+POD2MAN             := /usr/bin/pod2man
+POD2HTML            := /usr/bin/pod2html
+POD2TEXT            := /usr/bin/pod2text
+PANDOC              := /usr/bin/pandoc
+PERL                := /usr/bin/perl
diff --git a/config/Paths.mk b/config/Paths.mk
new file mode 100644
index 0000000000..3ec193e50e
--- /dev/null
+++ b/config/Paths.mk
@@ -0,0 +1,61 @@
+# Xen system configuration
+# ========================
+#
+# Xen uses a set of variables for system configuration and at build time,
+# because of this these variables are defined on one master input source file
+# and is generated after running ./configure. The master source is located
+# on the xen source tree at under config/Paths.mk.in and it is used to
+# generate shell or header files by the build system upon demand through the
+# use of the helper makefile helper buildmakevars2file().
+#
+# For more documentation you can refer to the wiki:
+#
+# http://wiki.xen.org/wiki/Category:Host_Configuration#System_wide_xen_configuration
+
+PACKAGE_TARNAME          := xen
+prefix                   := /usr
+bindir                   := /usr/bin
+sbindir                  := /usr/sbin
+libdir                   := /usr/lib64
+libexecdir               := /usr/libexec
+datarootdir              := ${prefix}/share
+mandir                   := ${datarootdir}/man
+docdir                   := ${datarootdir}/doc/${PACKAGE_TARNAME}
+dvidir                   := ${docdir}
+htmldir                  := ${docdir}
+pdfdir                   := ${docdir}
+psdir                    := ${docdir}
+includedir               := ${prefix}/include
+localstatedir            := /var
+sysconfdir               := /etc
+
+LIBEXEC                  := /usr/libexec/xen
+LIBEXEC_BIN              := /usr/libexec/xen/bin
+LIBEXEC_LIB              := /usr/libexec/xen/lib
+LIBEXEC_INC              := /usr/libexec/xen/include
+
+SHAREDIR                 := /usr/share
+MAN1DIR                  := $(mandir)/man1
+MAN8DIR                  := $(mandir)/man8
+
+XEN_RUN_DIR              := /var/run/xen
+XEN_LOG_DIR              := /var/log/xen
+XEN_LIB_DIR              := /var/lib/xen
+XEN_LIB_STORED           := /var/lib/xenstored
+XEN_RUN_STORED           := /var/run/xenstored
+
+CONFIG_DIR               := /etc
+INITD_DIR                := /etc/rc.d/init.d
+CONFIG_LEAF_DIR          := sysconfig
+BASH_COMPLETION_DIR      := $(CONFIG_DIR)/bash_completion.d
+XEN_LOCK_DIR             := /var/lock
+XEN_PAGING_DIR           := /var/lib/xen/xenpaging
+XEN_DUMP_DIR             := /var/lib/xen/dump
+DEBUG_DIR                := /usr/lib/debug
+
+XENFIRMWAREDIR           := /usr/libexec/xen/boot
+
+XEN_CONFIG_DIR           := /etc/xen
+XEN_SCRIPT_DIR           := /etc/xen/scripts
+
+PKG_INSTALLDIR           := ${libdir}/pkgconfig
diff --git a/config/Stubdom.mk b/config/Stubdom.mk
new file mode 100644
index 0000000000..a8bb1d3147
--- /dev/null
+++ b/config/Stubdom.mk
@@ -0,0 +1,45 @@
+# Prefix and install folder
+prefix              := /usr/local
+PREFIX              := $(prefix)
+exec_prefix         := ${prefix}
+libdir              := ${exec_prefix}/lib
+LIBDIR              := $(libdir)
+
+# Path Programs
+CMAKE               := /usr/bin/cmake
+FETCHER             := /usr/bin/wget -c -O
+
+# A debug build of stubdom? //FIXME: Someone make this do something
+debug               := y
+vtpm = y
+
+STUBDOM_TARGETS     :=  ioemu grub xenstore vtpm vtpmmgr
+STUBDOM_BUILD       :=  ioemu-stubdom pv-grub xenstore-stubdom vtpm-stubdom vtpmmgr-stubdom
+STUBDOM_INSTALL     :=  install-ioemu install-grub install-xenstore install-vtpm install-vtpmmgr
+
+ZLIB_VERSION        := 1.2.3
+ZLIB_URL            := $(XEN_EXTFILES_URL)
+
+LIBPCI_VERSION      := 2.2.9
+LIBPCI_URL          := $(XEN_EXTFILES_URL)
+
+NEWLIB_VERSION      := 1.16.0
+NEWLIB_URL          := $(XEN_EXTFILES_URL)
+
+LWIP_VERSION        := 1.3.0
+LWIP_URL            := $(XEN_EXTFILES_URL)
+
+GRUB_VERSION        := 0.97
+GRUB_URL            := $(XEN_EXTFILES_URL)
+
+OCAML_VERSION       := 3.11.0
+OCAML_URL           := http://caml.inria.fr/pub/distrib/ocaml-3.11
+
+GMP_VERSION         := 4.3.2
+GMP_URL             := $(XEN_EXTFILES_URL)
+
+POLARSSL_VERSION    := 1.1.4
+POLARSSL_URL        := $(XEN_EXTFILES_URL)
+
+TPMEMU_VERSION      := 0.7.4
+TPMEMU_URL          := $(XEN_EXTFILES_URL)
diff --git a/config/Tools.mk b/config/Tools.mk
new file mode 100644
index 0000000000..f07ff6a5c0
--- /dev/null
+++ b/config/Tools.mk
@@ -0,0 +1,78 @@
+-include $(XEN_ROOT)/config/Paths.mk
+
+CONFIG_RUMP         := n
+ifeq ($(CONFIG_RUMP),y)
+XEN_OS              := NetBSDRump
+endif
+
+# Tools path
+BISON               := 
+FLEX                := 
+PYTHON              := python
+PYTHON_PATH         := /usr/bin/python
+PY_NOOPT_CFLAGS     := -O1
+PERL                := /usr/bin/perl
+BASH                := /bin/sh
+XGETTTEXT           := /usr/bin/xgettext
+AS86                := /usr/bin/as86
+LD86                := /usr/bin/ld86
+BCC                 := /usr/bin/bcc
+IASL                := /usr/bin/iasl
+AWK                 := /usr/bin/awk
+FETCHER             := /usr/bin/false
+
+# Extra folder for libs/includes
+PREPEND_INCLUDES    := 
+PREPEND_LIB         := 
+APPEND_INCLUDES     := 
+APPEND_LIB          := 
+
+PTHREAD_CFLAGS      := -pthread
+PTHREAD_LDFLAGS     := -pthread
+PTHREAD_LIBS        := 
+
+PTYFUNCS_LIBS       := -lutil
+
+LIBNL3_LIBS         := 
+LIBNL3_CFLAGS       := 
+XEN_TOOLS_RPATH     := n
+
+# Download GIT repositories via HTTP or GIT's own protocol?
+# GIT's protocol is faster and more robust, when it works at all (firewalls
+# may block it). We make it the default, but if your GIT repository downloads
+# fail or hang, please pass --enable-githttp to configure.
+GIT_HTTP            ?= n
+
+# Optional components
+XENSTAT_XENTOP      := y
+OCAML_TOOLS         := y
+FLASK_POLICY        := n
+CONFIG_OVMF         := n
+CONFIG_ROMBIOS      := y
+CONFIG_SEABIOS      := n
+CONFIG_IPXE         := n
+CONFIG_QEMU_TRAD    := n
+CONFIG_QEMU_XEN     := n
+CONFIG_QEMUU_EXTRA_ARGS:= 
+CONFIG_LIBNL        := n
+
+CONFIG_SYSTEMD      := y
+SYSTEMD_CFLAGS      :=  
+SYSTEMD_LIBS        := -lsystemd  
+XEN_SYSTEMD_DIR     := $(prefix)/lib/systemd/system/
+XEN_SYSTEMD_MODULES_LOAD := $(prefix)/lib/modules-load.d/
+CONFIG_9PFS         := 
+
+LINUX_BACKEND_MODULES := xen-evtchn xen-gntdev xen-gntalloc xen-blkback xen-netback xen-pciback evtchn gntdev netbk blkbk xen-scsibk usbbk pciback xen-acpi-processor
+
+#System options
+ZLIB                :=  -DHAVE_BZLIB -lbz2 -DHAVE_LZMA -llzma -DHAVE_LZO1X -llzo2
+CONFIG_LIBICONV     := n
+EXTFS_LIBS          := -lext2fs
+CURSES_LIBS         := -lncurses
+TINFO_LIBS          := -ltinfo
+ARGP_LDFLAGS        := 
+
+FILE_OFFSET_BITS    := 
+
+CONFIG_PV_SHIM      := n
diff --git a/config/Toplevel.mk b/config/Toplevel.mk
new file mode 100644
index 0000000000..6e90a465b9
--- /dev/null
+++ b/config/Toplevel.mk
@@ -0,0 +1,2 @@
+SUBSYSTEMS               :=  xen tools docs
+GIT_HTTP                 ?= n
diff --git a/tools/config.h b/tools/config.h
new file mode 100644
index 0000000000..5006dfbd5b
--- /dev/null
+++ b/tools/config.h
@@ -0,0 +1,139 @@
+/* config.h.  Generated from config.h.in by configure.  */
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Enabling support partial device tree in libxl */
+/* #undef ENABLE_PARTIAL_DEVICE_TREE */
+
+/* Define to 1 if you have the declaration of `fdt_first_subnode', and to 0 if
+   you don't. */
+/* #undef HAVE_DECL_FDT_FIRST_SUBNODE */
+
+/* Define to 1 if you have the declaration of `fdt_next_subnode', and to 0 if
+   you don't. */
+/* #undef HAVE_DECL_FDT_NEXT_SUBNODE */
+
+/* Define to 1 if you have the declaration of `fdt_property_u32', and to 0 if
+   you don't. */
+/* #undef HAVE_DECL_FDT_PROPERTY_U32 */
+
+/* Define to 1 if you have the `fdt_first_subnode' function. */
+/* #undef HAVE_FDT_FIRST_SUBNODE */
+
+/* Define to 1 if you have the `fdt_next_subnode' function. */
+/* #undef HAVE_FDT_NEXT_SUBNODE */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the `fdt' library (-lfdt). */
+/* #undef HAVE_LIBFDT */
+
+/* Define to 1 if you have the `lzma' library (-llzma). */
+#define HAVE_LIBLZMA 1
+
+/* Define to 1 if you have the `yajl' library (-lyajl). */
+#define HAVE_LIBYAJL 1
+
+/* Define to 1 if you have the `z' library (-lz). */
+#define HAVE_LIBZ 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Qemu traditional enabled */
+/* #undef HAVE_QEMU_TRADITIONAL */
+
+/* ROMBIOS enabled */
+#define HAVE_ROMBIOS 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#define HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Systemd available and enabled */
+#define HAVE_SYSTEMD 1
+
+/* Define to 1 if you have the <sys/eventfd.h> header file. */
+#define HAVE_SYS_EVENTFD_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the <utmp.h> header file. */
+#define HAVE_UTMP_H 1
+
+/* Define to 1 if you have the <valgrind/memcheck.h> header file. */
+/* #undef HAVE_VALGRIND_MEMCHECK_H */
+
+/* Define to 1 if you have the <yajl/yajl_version.h> header file. */
+#define HAVE_YAJL_YAJL_VERSION_H 1
+
+/* Define curses header to use */
+#define INCLUDE_CURSES_H <ncurses.h>
+
+/* Define extfs header to use */
+#define INCLUDE_EXTFS_H <ext2fs/ext2fs.h>
+
+/* Define json header to use */
+#define INCLUDE_JSON_H <json/json.h>
+
+/* libutil header file name */
+/* #undef INCLUDE_LIBUTIL_H */
+
+/* IPXE path */
+#define IPXE_PATH "/usr/share/ipxe/ipxe.bin"
+
+/* OVMF path */
+#define OVMF_PATH "/usr/share/edk2/OVMF.fd"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "xen-devel@lists.xen.org"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Xen Hypervisor Tools"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Xen Hypervisor Tools 4.13"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "xen"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL "http://www.xen.org/"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "4.13"
+
+/* Qemu Xen path */
+#define QEMU_XEN_PATH "/usr/lib64/xen/bin/qemu-system-i386"
+
+/* SeaBIOS path */
+/* #undef SEABIOS_PATH */
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Enable large inode numbers on Mac OS X 10.5.  */
+#ifndef _DARWIN_USE_64_BIT_INODE
+# define _DARWIN_USE_64_BIT_INODE 1
+#endif
+
+/* Number of bits in a file offset, on hosts where this is settable. */
+/* #undef _FILE_OFFSET_BITS */
+
+/* Define for large files, on AIX-style hosts. */
+/* #undef _LARGE_FILES */
diff --git a/tools/ocaml/xenstored/oxenstored.conf.in b/tools/ocaml/xenstored/oxenstored.conf.in
index 151b65b72d..74f9cbca98 100644
--- a/tools/ocaml/xenstored/oxenstored.conf.in
+++ b/tools/ocaml/xenstored/oxenstored.conf.in
@@ -46,9 +46,9 @@ perms-activate = true
 
 # Activate quota
 quota-activate = true
-quota-maxentity = 1000
+quota-maxentity = 8192
 quota-maxsize = 2048
-quota-maxwatch = 100
+quota-maxwatch = 512
 quota-transaction = 10
 quota-maxrequests = 1024
 
@@ -59,12 +59,14 @@ persistent = false
 # xenstored-log-file = @XEN_LOG_DIR@/xenstored.log
 # xenstored-log-level = null
 # xenstored-log-nb-files = 10
+xenstored-log-file = syslog:daemon
 
 # Xenstored access logs
 # access-log-file = @XEN_LOG_DIR@/xenstored-access.log
 # access-log-nb-lines = 13215
 # acesss-log-nb-chars = 180
 # access-log-special-ops = false
+access-log-file = syslog:local3
 
 # Perodically scanning all the rings as a safenet for lazy clients.
 # Define the interval in seconds, set to negative to disable.
diff --git a/xs-reconfigure.sh b/xs-reconfigure.sh
new file mode 100755
index 0000000000..17d7e521c7
--- /dev/null
+++ b/xs-reconfigure.sh
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+./configure --prefix=/usr \
+            --libdir=/usr/lib64 \
+            --libexecdir=/usr/libexec \
+            --disable-qemu-traditional \
+            --disable-seabios \
+            --disable-stubdom \
+            --disable-xsmpolicy \
+            --disable-pvshim \
+            --enable-rombios \
+            --enable-systemd \
+            --with-xenstored=oxenstored \
+            --with-system-qemu=/usr/lib64/xen/bin/qemu-system-i386 \
+            --with-system-ipxe=/usr/share/ipxe/ipxe.bin \
+            --with-system-ovmf=/usr/share/edk2/OVMF.fd
+
+shopt -s extglob
+
+# Pick all config-$foo but skip *.old
+for CFG in buildconfigs/config-*!(.old)
+do
+    make -C xen/ KCONFIG_CONFIG=../$CFG olddefconfig
+done
